### PR TITLE
Support CSS `field-sizing` on text-area, text-field, number-field, and combobox

### LINKED ISSUE #1228: Make textarea height optionally auto-resizable
## 😯 Problem to Solve

It would be nice if nimble-text-area offered auto-resize like [Angular Material](https://material.angular.io/components/input/overview#auto-resizing-textarea-elements) and [PrimeNG](https://primeng.org/inputtextarea#autoresize). This would allow the input to grow vertically vs. show scrollbars. See [this example](https://stackblitz.com/edit/angular-ivy-pcaccy?file=src/app/app.component.ts) implemented with a contenteditable div:

textarea

## 💁 Proposed Solution

Add an attribute like `[autoResize]="true"`

### PR DESCRIPTION
**Pull request by m-akinc**
# Pull Request

## 🤨 Rationale

I wanted to size a `nimble-text-field` to the width of its content. This isn't straightforward unless you use the [`field-sizing` CSS property](https://developer.mozilla.org/en-US/docs/Web/CSS/Reference/Properties/field-sizing). The property has been supported in Chrome/Edge for a couple years, but only in Safari (12/2025) and Firefox (6/2026) fairly recently. It would be very useful if you could apply the style to our custom elements and see the effect on the the native elements in the shadow DOM.

Also, apparently resolves #1228

Found one edge case (field-sizing: content, and a fixed width) captured in matrix tests that does not block this PR. Created an issue: https://github.com/ni/nimble/issues/2999

## 👩‍💻 Implementation

`field-sizing` is only supported on `input` and `textarea` elements, so it only really applies to `nimble-text-field`, `nimble-number-field`, `nimble-text-area`, and `nimble-combo-box`.

Updated stylesheets to appy `field-sizing: inherit` to the native `input` and `textarea` elements, as well as their containing `div`s and/or `slot`s. This allows a user to style the host element and have that style "cascade".

In the case of `nimble-combobox` and `nimble-text-area`, there is a `min-width` by default, so if you want it to size smaller than that, you need to explicitly change that property, too.

Additionally, added a 4-px padding to the right side of the `nimble-number-field` when the stepper buttons are hidden to better match the design.

## 🧪 Testing

Added visual test cases for `nimble-text-field`, `nimble-number-field`, `nimble-text-area`, and `nimble-combobox`.

## ✅ Checklist
- [x] I have updated the project documentation to reflect my changes or determined no changes are needed.

### CODE REVIEW

File: packages/nimble-components/src/number-field/styles.ts
Comment (rajsite) [inline packages/nimble-components/src/number-field/styles.ts:30]: Can you set a default for field-sizing on the host to make it clearer to follow the inherit path. It's a bit subtle to follow the intended behavior otherwise.

File: change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (rajsite) [inline change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: combobox is also an input text style control

Comment (m-akinc) [reply change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: Added support to combobox, too, though we give combobox a `min-width` by default, so if you want it to go below that size, you have to also change `min-width`.

File: packages/nimble-components/src/number-field/styles.ts
Comment (m-akinc) [reply packages/nimble-components/src/number-field/styles.ts:30]: Done

File: change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json (hunk 1 above)
Comment (rajsite) [inline change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: talked offline, high-level concerns:
- Want to avoid any changes to default sizing behvior, i.e. in some iterations of the PR the size attr on input or width of parts of the control via css are changing. Don't want to make changes that are observable to clients.
- During demos point was brought up about controls that render additional content inside them, i.e. text-field action buttons, error visible states, those should be exercised in stories
- This behavior is an explicit feature request, we should capture styling docs in the control's respective storybook pages bout enabling this style and close the related issue https://github.com/ni/nimble/issues/1228

Comment (rajsite) [inline change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: @m-akinc  As an fyi, you can target the part directly today to enable field-sizing: https://github.com/ni/nimble/issues/1228#issuecomment-2705169585

But the changes in this PR would certainly be nice (just saying in-case it's blocking something to shore-up these questions)

Comment (m-akinc) [reply change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: I can wait, I think.

Comment (m-akinc) [reply change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: Done

File: packages/storybook/src/nimble/text-area/text-area.mdx [hunk 2]
```diff
@@ -8,12 +8,16 @@ import ComponentApisLink from '../../docs/component-apis-link.mdx';
...
+If you configure your text area to be user-resizable (with the `resize`
+attribute) in a certain dimension, do not set an explicit size for that dimension
+(via `height` and/or `width` `style` properties), or you may experience unexpected
 resize behavior. If you want to set the initial size of a resizable text area,
 use the `rows` and/or `cols` attribute(s).
 
+To make the text area grow automatically to fit its content, style the element
+with `field-sizing: content`. This will allow it to grow both horizontally and
+vertically. Set a fixed `width` if you only want it to grow vertically.
+
```
Comment (rajsite) [inline packages/storybook/src/nimble/text-area/text-area.mdx]: I think we should document this for each component being changed, not just textarea. Can you make a shared doc component to use across each of the elements we are adding support for?

Can make a shared styling doc in the pattern of button-styling-docs.mdx: https://github.com/ni/nimble/blame/main/packages/storybook/src/nimble/patterns/button/button-styling-docs.mdx

But don't have to go that overkill with examples and everything. Takeaways are a shared mdx for that doc snippet that ends up under the styling section of each of the component docs.

Comment (m-akinc) [reply packages/storybook/src/nimble/text-area/text-area.mdx]: Done

File: packages/nimble-components/src/number-field/styles.ts [hunk 3]
```diff
@@ -131,12 +133,17 @@ export const styles = css`
...
         height: var(--ni-private-height-within-border);
         width: 100%;
         border: none;
         padding: 0px;
         padding-left: ${mediumPadding};
     }
+    
+    :host([hide-step]) .control {
+        padding-right: ${mediumPadding};
+    }
```
Comment (rajsite) [inline packages/nimble-components/src/number-field/styles.ts:146]: This seems like a completely unrelated change

It looks like the following in this PR (medium right padding):

Image

This way on main (no right padding):

Image

Niether of which align with the design:
image

File: change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json (hunk 1 above)
Comment (rajsite) [inline change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: PR feedback:
- The padding change should be documented in PR and change since it is an unrelated change and differs from main (no right padding)
- The padding with hide-step should be small padding to match the text field in figma, i.e. a numeric text field with hide-step and text field in line with each other with long values should fill the same space:
    Image
    [figma ref](https://www.figma.com/design/PO9mFOu5BCl8aJvFchEeuN/Nimble_Components?node-id=1295-51200&m=dev)
- lets make these as matrix tests instead of hand rolling the dimensions
- text field should have a long text dimension added like the others (it jsut has short)

File: packages/nimble-components/src/number-field/styles.ts (hunk 3 above)
Comment (rajsite) [reply packages/nimble-components/src/number-field/styles.ts:146]: merged with other comment

File: change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json (hunk 1 above)
Comment (m-akinc) [reply change/@ni-nimble-components-6d58234e-1827-4414-a92d-2c3b47400e14.json:1]: Done

**Review by rajsite — APPROVED** [review]
